runtime/src/main/java/org/corfudb/runtime/collections/StreamListener.java

```diff
@@ -7,6 +7,7 @@
  */
 public interface StreamListener {
 
+    // TODO(Maithem): what is this?? remove?
```

Comment:
Seems like this API already leaked to the Application, can't remove it without breaking their code.

Reply:
actually this is not intended to be consumed externally, the onNext is the clients callback

Reply:
Yes, I was under the assumption that it was used. Turns out it is note. Just removed it.

Reply:
mmm but no guys we can't just remove it, that breaks the StreamListenerResumePolicy class, this pattern requires that on every onNext callback completion, we capture the last processed entry, so in case of any errors we can automatically attempt to re-subscribe from the last processed entry (onNext processed). With this change there is no guarantee that the parent's onNext (i..e, the one in StreamListenerResumePolicy is going to be called after the client's onNext). For that reason, onNextEntry was the internal method used, which called client's onNext + any specific logic required by the patterns.

Reply:
It did effectively break tests in: StreamingPatternsIT. We can discuss this offline, but given that it's unrelated to this change, we can also leave it for later and focus on the actual change of the Streaming API.

Reply:
`StreamListenerResumePolicy` is an abstract class, concrete classes that implement it need to call super.onNext. I don't think we should keep in in the public API just to relay this "inheritance"/common code. Like this: 
https://github.com/CorfuDB/CorfuDB/pull/3047/files#diff-76c7fea3be840598202abc5bbd92ed59c484ff0e9e5e8811ed5faf1fea50872aR92

Reply:
Reverted.